### ISSUE TITLE
/nickserv setusername

### ISSUE THREAD

**Issue by slingamn** [oragono/oragono#1162]
We got a report in #oragono from someone who doesn't like that the username of always-on clients is always `~user` (after a server restart).

Let's provide an additional nickserv command to change the username (and send chghost). To avoid creating any incentive to use ident, this should always prefix the chosen username with `~`.

**Comment by slingamn** [comment]
I started doing this and realized I really don't want to. User/ident is cancelled.

**Comment by Amiga60077** [comment]
Well it could simply be changed to the users actual nickname, so  Amiga600!~user@host - would become Amiga600!~Amiga600@host instead?

**Comment by slingamn** [comment]
Non-ASCII nicks are invalid as user/ident fields.

We've actually gone even more aggressively in the opposite direction here with #1213 (an option to always coerce user/ident to `~user`, even for non-always-on clients). I'm tempted to make that a recommended default, actually, since it makes all implementations of `/kb` function identically.

**Comment by Amiga60077** [comment]
Well perhaps have logged in users show as Amiga600!Amiga600@host then? - Just forget using the ~ character perhaps?

**Comment by slingamn** [comment]
It's more like, `姚明` is a valid nickname in the default configuration of oragono, but it's not valid as a user/ident (with or without a preceding `~`).

**Comment by Amiga60077** [comment]
Surely, that could be overcome with a simple check
ie. remove all non-ascii chars - if string has still illegal chars, set to "user" - if not set to "nickname"

**Comment by slingamn** [comment]
To what end, though?

#1213 appealed to me so much because the people who asked for it reported that user/ident had become merely a source of confusion. I don't really see the point of the user/ident field anymore.

**Comment by DanielOaks** [comment]
if the issue is aesthetics( / not including an english word) maybe just `dan!~@localhost` could look a lil nicer on the eyes than `~user` everywhere~?

**Comment by Amiga60077** [comment]
I wouldnt say its aesthetics its keeping to the IRC standards: https://en.wikipedia.org/wiki/Ident_protocol

Its hardly ever replied to for say Windows IRC Clients though, or people behind firewalls, as usually the IDENTD port is blocked by said firewall, in which case it is usually set to the nickname

For example on one irc server: Amiga600 is ~Amiga600@host
on a completely different one: Amiga600 is Amiga600@host

One simply ignores it, and passes my nickname onto IDENTD - the other correctly identifies me and puts my name without the ~ character there

**Comment by slingamn** [comment]
Ident lookups are already disabled in our default configuration, though.

**Comment by DanielOaks** [comment]
Yeah we're definitely aware of the standard way of doing things, the always-on way of working deviates from that a lot already though. In the 'default' config (the one where always-on, chat history, etc is all enabled), prioritising the account name (and making the nick and account name the same thing) is a pretty big part of sling's preference. The 'conventional' config's the one I like to keep as close to standard, traditional IRC though.

In something like the always-on mode, I can definitely see why sling'd prefer the ident field to never change. Letting users have their nickname as their ident would be nice for some users, but for people who aren't as used to traditional IRC they'll probably get confused as to why their nickname is used in two places.

In the channel an idea that was brought up was having an `coerce-ident` string instead of a `suppress-ident` bool, so the network admin can set the ident field to whatever string they wanted (with the default being `~` in the default config, and the conventional config not using it at all). For normal IRC, separate nick and ident fields are a given, but for the always-on side I reckon a change that de-emphasises the ident field makes sense, given that's also aimed as people less familiar with irc's intricacies.

**Comment by Amiga60077** [comment]
I guess for simplicity then, perhaps have a config option, so the end user can then either set it to what they want?
